### PR TITLE
merge bitcoin#19817, bitcoin#20470, bitcoin#22993: macOS C++17 bump

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: depends/hosts/darwin.mk [hunk 1]
```diff
@@ -60,16 +60,11 @@ $(foreach TOOL,$(cctools_TOOLS),$(eval darwin_$(TOOL) = $$(build_prefix)/bin/$$(
 #         Explicitly point to our binaries (e.g. cctools) so that they are
 #         ensured to be found and preferred over other possibilities.
 #
-#     -stdlib=libc++ -nostdinc++ -Xclang -cxx-isystem$(OSX_SDK)/usr/include/c++/v1
+#     -stdlib=libc++ -stdlib++-isystem$(OSX_SDK)/usr/include/c++/v1
```
Comment (PastaPastaPasta) [inline depends/hosts/darwin.mk:63]: 19817 missing changes from https://github.com/bitcoin/bitcoin/pull/19817/files#diff-964a007db35da98be548991cbc8815cc400d698508eb4b203921254e5c1bf8baR2-R5 (darwin.mk)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
one thing, also not sure why ci is unappy, otherwise looks dandy

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -37,8 +37,8 @@ builder-image:
   image: $CI_REGISTRY_IMAGE:builder-$CI_COMMIT_REF_SLUG
   variables:
     SDK_URL: https://bitcoincore.org/depends-sources/sdks
-    XCODE_VERSION: 11.3.1
-    XCODE_BUILD_ID: 11C505
+    XCODE_VERSION: 12.1
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply .gitlab-ci.yml]: Thanks for the fix! Odd how CI would behave like that 🙁

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**